File: 한지웅/Golang/삼각형.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
)

var reader *bufio.Reader = bufio.NewReader(os.Stdin)
var writer *bufio.Writer = bufio.NewWriter(os.Stdout)

func main() {
	defer writer.Flush()
	for {
		if cycle() == false {
			return
		}
	}
}

func cycle() bool {
	// -----cycle내에 reader, writer, Flush 존재 => 오답------
	// var reader *bufio.Reader = bufio.NewReader(os.Stdin)
	// var writer *bufio.Writer = bufio.NewWriter(os.Stdout)
	// defer writer.Flush()
	var a, b, c int

	fmt.Fscanln(reader, &a, &b, &c)
	if a == 0 && b == 0 && c == 0 {
		return false
	}
	if check_valid(a, b, c) == false {
		fmt.Println("Invalid")
	} else {
		if a == b && b == c {
			fmt.Println("Equilateral")
		} else if a != b && b != c && a != c {
			fmt.Println("Scalene")
		} else {
			fmt.Println("Isosceles")
		}
	}
	return true
}

func check_valid(a, b, c int) bool {
	pivot := []int{a, b, c}
	sort.Ints(pivot)
	if pivot[0]+pivot[1] <= pivot[2] {
		return false
	} else if pivot[0] == 0 || pivot[1] == 0 || pivot[2] == 0 {
		return false
	} else {
		return true
	}

}

File: 한지웅/삼각형.js
const fs = require('fs');
const input = fs.readFileSync('../input.txt').toString().trim().split("\n");
// const input = fs.readFileSync('/dev/stdin').toString().trim().split("\n");
// input의 개행문제 제거 작업 진행
for (i = 0; i < input.length - 1; i++) {
    input[i] = input[i].trim();
    var pivot = input[i].split(' ');
    var a = Number(pivot[0]);
    var b = Number(pivot[1]);
    var c = Number(pivot[2]);

    var parse = [a,b,c]
    parse.sort()


    var y = parse[0]
    var x = parse[1]
    var z = parse[2]

    if (a + b <= c || a === 0 || b === 0 || c === 0) {
        // 우선 삼각의 자격이 아닌지 확인
        console.log('Invalid')
    } else {
        if (a === b && b === c) {
            console.log('Equilateral')
        } else if (a === b || b === c || a === c) {
            console.log('Isosceles')
        } else {
            console.log('Scalene')
        }
    }
}

